src/main/scala/io/iohk/ethereum/network/PeerManagerActor.scala

```diff
@@ -227,4 +267,42 @@ object PeerManagerActor {
   }
 
   case class SendMessage(message: MessageSerializable, peerId: PeerId)
+
+  class PeerManagerState(var pendingIncomingPeers: Map[PeerId, Peer], var peers: Map[PeerId, Peer]) {
```

Comment:
I know this is only accessed within the Actor but I would prefer not to use `var`

Reply:
This class only purpose is to encapsulate state  operations of `PeerManagerActor` and we encode actor state operations in all our actors via var`s or mutable collections.
Do you prefer mutable collections or you mean to re-design this actor to mutate all state via `context.become` (theoretically it most pure way, but I am not convinced it is worth it for just one actor)

Reply:
I was thinking about `context.become` but you have a point that might not be worthy. 

What do you think @LukasGasior1 ?

Reply:
@AlanVerbner what do you think about adding `TODO` and task on youtrack to refactor this Actor to more pure style, as i think it would be good thing but also it exceeds the scope of this particular pr ?

Reply:
@KonradStaniec That's a good idea.